aws/validators.go

```diff
@@ -1580,6 +1580,19 @@ func validateAwsSSMMaintenanceWindowTaskName(v interface{}, k string) (ws []stri
 	return
 }
 
+func validateAwsSSMMaintenanceWindowTargetName(v interface{}, k string) (ws []string, errors []error) {
+	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_RegisterTaskWithMaintenanceWindow.html#systemsmanager-RegisterTaskWithMaintenanceWindow-request-Name
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-zA-Z0-9_\-.]{3,128}$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf(
+			"Only alphanumeric characters, hyphens, dots & underscores allowed in %q: %q (Must satisfy regular expression pattern: ^[a-zA-Z0-9_\\-.]{3,128}$)",
```

Comment:
I think we can shorten this message to 

```
errors = append(errors, fmt.Errorf("%q is not a valid value for %q (only alphanumeric characters, dots, underscores and hyphens are allowed)", value, k))
```

Reply:
Actually looking at this function a little further it looks like you can just use the function `validateAwsSSMName` defined on line 1157 and update the error message there.